src/admin/dumb_components/custom_email_form.js

```diff
@@ -1,19 +1,22 @@
 const React = require('react')
-const { map, objOf, zipWith, merge, compose } = require('ramda')
+const { map, objOf, zipWith, merge, compose, indexBy, prop, converge, dissoc } = require('ramda')
 
 export default ({ members, submit }) => {
   const onSubmit = (e) => {
     e.preventDefault()
-    const template = format_message(e.target[0].value)
-    const emailBodies = map(compose(objOf('email_body'), template), members)
-    const emails = zipWith(merge, members, emailBodies)
-    submit(emails)
+    const template = format_message(e.target)
+    const emailBodies = map(compose(objOf('content'), template), members)
+    const emailsArr = zipWith(merge, members, emailBodies)
+    const setEmailKey = map(compose(indexBy, prop), ['primary_email', 'secondary_email'])
```

Comment:
For custom email, the email body gets typed , then they will click 'send'
which sends the body to the reducer to be formatted and then needs to be
sent to the server. So one action to send the body to reducer, and one to
send the formatted emails obj to the server.

Unless we bundle up the formatting inside the action?

On Thu, Jul 14, 2016 at 12:16 PM, Rafe <notifications@github.com> wrote:

> In src/admin/dumb_components/custom_email_form.js
> <https://github.com/foundersandcoders/sail-back/pull/286#discussion_r70787969>
> :
>
> >
> >  export default ({ members, submit }) => {
> >    const onSubmit = (e) => {
> >      e.preventDefault()
> > -    const template = format_message(e.target[0].value)
> > -    const emailBodies = map(compose(objOf('email_body'), template), members)
> > -    const emails = zipWith(merge, members, emailBodies)
> > -    submit(emails)
> > +    const template = format_message(e.target)
> > +    const emailBodies = map(compose(objOf('content'), template), members)
> > +    const emailsArr = zipWith(merge, members, emailBodies)
> > +    const setEmailKey = map(compose(indexBy, prop), ['primary_email', 'secondary_email'])
>
> It's not clear to me why you'd need to call 2 actions?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/foundersandcoders/sail-back/pull/286/files/02e081a100a86bfc84019925babe1478a6dd75db#r70787969>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AMDYNQBXkZv-en-wvtKxXuu4Ke2Dp-6Vks5qVhqKgaJpZM4JKksx>
> .
>